File: src/test/java/io/binac/leetcode/MinimumWindowSubstringTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MinimumWindowSubstringTest {
    private final MinimumWindowSubstring.Solution1 solution1 = new MinimumWindowSubstring.Solution1();

    @Test
    void test1() {
        assertEquals("BANC", solution1.minWindow("ADOBECODEBANC", "ABC"));
    }
}

File: cpp/tests/MinimumWindowSubstringTest.cpp
#include "MinimumWindowSubstring.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(MinimumWindowSubstring, Solution76_1) {
  auto S76_1 = Solution76_1();

  EXPECT_EQ("BANC", S76_1.minWindow("ADOBECODEBANC", "ABC"));
  EXPECT_EQ("a", S76_1.minWindow("a", "a"));
  EXPECT_EQ("", S76_1.minWindow("a", "aa"));
}